src/score_hv/harvesters/daily_bfg.py

```diff
@@ -141,107 +231,127 @@ def get_data(self):
                                        combine='nested', 
                                        concat_dim='time',
                                        decode_times=True)
-        gridcell_area_data = xr.open_dataset(get_gridcell_area_data_path())
-        gridcell_area_weights = gridcell_area_data.variables['area']
-
-        temporal_endpoints = np.array([cftime.date2num(time,
-            'hours since 1951-01-01 00:00:00') for time in xr_dataset['time']])
+        latitudes=xr_dataset['grid_yt']
+        longitudes=xr_dataset['grid_xt']
+        gridcell_area_data=xr.open_dataset(get_gridcell_area_data_path())
+        gridcell_area_weights=(gridcell_area_data.variables['area'])
+        num_lat = len(latitudes.values)
+        num_lon = len(longitudes.values)
+        total_original_elements = num_lat * num_lon
+        total_regional_elements = total_original_elements
+        check_weights(gridcell_area_weights,total_original_elements,total_regional_elements) 
         
-        if len(self.config.harvest_filenames) > 1:
-            """ can estimate the time step only if there're more than 1 
-                timestamps
+        regions_instance=GeoRegions(latitudes,longitudes)
+        user_regions = self.config.region
+        if user_regions:
+           """
+             The user has requested a region/regions so we need to
+             instantiate the region class.  We do this by giving the
+             entire lat and lon values in the data set.
+             """
+           num_regions   = len(user_regions)
+           for region_info in user_regions:
+                    region_name, min_lat, max_lat, east_lon, west_lon = region_info
+                    regions_instance.add_user_region(region_name,min_lat,max_lat,east_lon,west_lon)
+        else:
             """
-            temporal_midpoints = temporal_endpoints - np.gradient(temporal_endpoints)/2.
+              The user has not requested a specific region.  So
+              we will set a global region.  This makes the code
+              much simplier.
+              """
+            region_name="global"
+            min_lat=-90.0
+            max_lat=90.0
+            east_lon=0.0
+            west_lon=360.0
+            regions_instance.add_user_region(region_name,min_lat,max_lat,east_lon,west_lon)
```

Comment:
```suggestion
            regions_catalog.add_user_region(DEFAULT_REGION)
```
remove above definitions (min_lat=, max_lat=, etc...) as needed